lookout/style/typos/research/__main__.py

```diff
@@ -1,44 +1,104 @@
 
+from lookout.style.typos.research.baseline import baseline
 from lookout.style.typos.research.nn_prediction import DEFAULT_BATCH_SIZE, DEFAULT_DECAY, \
     DEFAULT_LR, DEFAULT_NUM_EPOCHS, DEFAULT_NUM_NEURONS, create_and_train_nn_prediction_from_file
 from lookout.style.typos.utils import CORRECT_TOKEN_COLUMN, TYPO_COLUMN
 
 
 def create_parser():
     parser = argparse.ArgumentParser(formatter_class=ArgumentDefaultsHelpFormatterNoNone,
-                                     description="Train nn model for correction embedding"
-                                                 "prediction.")
-    parser.add_argument("--fasttext", type=str, required=True,
-                        help="Path to the binary dump of a FastText model.")
-    parser.add_argument("--data", type=str, required=True,
-                        help="Path to a CSV dump of pandas.DataFrame containing"
-                             "columns [%s, %s]" % (CORRECT_TOKEN_COLUMN, TYPO_COLUMN))
-    parser.add_argument("--dump", type=str,
-                        help="Path to the file where to dump the trained NN model.")
-    parser.add_argument("--num-neurons", type=int, help="Number of neurons in each hidden layer.",
-                        default=DEFAULT_NUM_NEURONS)
-    parser.add_argument("--batch-size", type=int, default=DEFAULT_BATCH_SIZE,
-                        help="Batch size for training.")
-    parser.add_argument("--lr", type=float, default=DEFAULT_LR, help="Learning rate.")
-    parser.add_argument("--decay", type=float, default=DEFAULT_DECAY,
-                        help="Learning rate exponential decay per epoch.")
-    parser.add_argument("--num-epochs", type=int, default=DEFAULT_NUM_EPOCHS,
-                        help="Number of training passes over the dataset.")
-    parser.add_argument("--log-level", default="INFO", choices=logging._nameToLevel,
-                        help="Logging verbosity.")
+                                     description='Typos correction tools')
+    subparsers = parser.add_subparsers(help="Commands", dest="command")
+
+    def add_parser(name, help_message):
+        return subparsers.add_parser(name, help=help_message,
+                                     formatter_class=ArgumentDefaultsHelpFormatterNoNone)
+
+    # ------------------------------------------------------------------------
+    baseline_parser = add_parser("baseline", """
+    Typos correction model, based on SymSpell lookout algorithm
+
+    https://github.com/wolfgarbe/SymSpell
+
+    and simple Random Forest classifier, based on token frequencies
+    and edit distance between typo and candidate.
+
+    May use pretrained model.
+
+    Dumps model if the dump model file specified.
+
+    Training: Requires token frequencies info and training dataframe,
+              may also accept dataframe with precalculated candidates suggestions.
+
+    Testing:  Requires training first or pretrained model, testing dataframe,
```

Comment:
```suggestion
    Testing:  Requires a pretrained model, testing dataframe,
```

Let's KISS and always require a trained model for evaluation

Reply:
testing dataframe?